packages/events/Subscriber/index.js

```diff
@@ -11,12 +13,35 @@ class Subscriber {
     this.unsubscribeListeners = {};
     this.quiet = config.quiet;
     this.config = config;
-    if (config.logger) this.logger = config.logger;
-    if (initialization) initialization.bind(this)(config);
+
+    Object.defineProperty(this, "logger", {
+      get: () => {
+        if (!this._logger) {
+          this.logger = this.config.logger || console;
```

Comment:
note: this line is intentionally `this.logger =` and not `this._logger =`, as I wanted to invoke the wrapper logic in the setter.